libvast/vast/word.hpp

```diff
@@ -314,22 +271,14 @@ static constexpr auto rank(T x)
 /// @param i The position up to where to count.
 /// @returns *rank_i(x)*.
 /// @pre `i < width`
-template <bool Bit = true, class T>
-static constexpr auto rank(T x, detail::word_size_type<T> i)
--> std::enable_if_t<
-  Bit && detail::is_unsigned_integral_v<T>,
-  detail::word_size_type<T>
-> {
+template <bool Bit = true, detail::unsigned_integral T>
+requires(Bit) static constexpr auto rank(T x, detail::word_size_type<T> i) {
```

Comment:
Is there some clang-format rule to put this in a line of its own or in the previous line?

Reply:
In my experience, the `clang-format` rules for `requires` expressions, concepts,  and `requires requires` is all pretty poor until llvm 12. I use llvm 12.0.1 locally, which may differ for clang-format of this function template (and others).

There aren't many knobs we can set in `.clang-format` yet. In llvm12, they introduced `BreakBeforeConceptDeclaration` which is `true` by default it seems. This is what it up doing locally for me.

If we update the CI compilers even more-so, I'm happy to add that to the `.clang-format` file. I doubt Bullseye is ready for llvm12/gcc11 yet though, otherwise I suspect we would have jumped right to those?

Reply:
It seems the style format CI job is failing due to the differences I mentioned (similar in vein to https://github.com/tenzir/vast/issues/1779).

Reply:
I have no issue with updating the style check workflow to use a newer clang-format regardless of the platforms we support. We can easily have higher requirements for developing than for running VAST.

I think the easiest way to fix this is to `brew install llvm` in that workflow to get latest llvm from Linuxbrew, which is pre-installed in the Ubuntu images in GitHub Actions.

Do you mind if I take this branch over and push the changes directly onto this?

Reply:
Sure, feel free to run with it for fixing up `clang-format` changes. I just disabled `clang-format` in `concepts.hpp` to avoid the error which should work, but it's not what I want to write. I want code to format concepts and requires etc nicely which comes mostly in `llvm12`. There are a few bugs with formatting in llvm trunk with concepts and friends, but none get hit in this PR.

Reply:
I've made the change, let's see if it runs through.

Reply:
You made a typo in `latest`.  If we fix that and remove the `clang-format off` section in `concepts.hpp` and it works, I'm good with it.

Otherwise, as-is with this PR, the `clang-format off` toggle in `concepts.hpp` is good enough to pass the style check right now.

Reply:
Eh, seems like there's something wrong with the clang-format bottle for Linuxbrew. I'll take another look in a few minutes.

Reply:
No worries. Feel free to do it as a follow-up PR and we can drop that commit for this branch.

Reply:
Installation seems to have worked. Can you take it over from here again?

Reply:
Yep. Just removed the explicit `clang-format off` toggle in `concepts.hpp`. We'll see if the style job passes now 🤞 .

Thanks for the help. 😄